### PR TITLE
Add docstrings for decorators.py

### PR DESCRIPTION
**Pull request by tpdorsey**
Adds docstring coverage for decorators.py

## Checklist

* [ ] Unit tests for the changes exist
* [ ] Tests pass on CI and coverage remains at 100%
* [ ] Documentation reflects the changes where applicable
* [ ] `changes/<pull request or issue id>-<github username>.md` file added describing change
  (see [changes/README.md](https://github.com/samuelcolvin/pydantic/blob/master/changes/README.md) for details)
* [ ] My PR is ready to review, **please add a comment including the phrase "please review" to assign reviewers**

### CODE REVIEW

File: pydantic/decorators.py [hunk 1]
```diff
@@ -168,6 +168,18 @@ def validator(
     mode: Literal['before', 'after'] = 'before' if pre is True else 'after'
 
     def dec(f: Any) -> _decorators.PydanticDescriptorProxy[Any]:
+        """
+        A decorator function for pydantic validation.
```
Comment (Kludex) [inline pydantic/decorators.py:172]: Can someone else suggest "what it does" instead of "what it is" before merging?

File: pydantic/decorators.py [hunk 2]
```diff
@@ -412,6 +454,17 @@ def field_serializer(
...
+        """
+        Decorator that creates a `PydanticDescriptorProxy` object.
+
+        Args:
+            f (Callable[..., Any] | staticmethod[Any, Any] | classmethod[Any, Any, Any]):
+                A callable object or a static/class method.
+
+        Returns:
+            _decorators.PydanticDescriptorProxy[Any]: A `PydanticDescriptorProxy` object.
+
```
Comment (Kludex) [inline pydantic/decorators.py:466]:
```suggestion
```

File: pydantic/decorators.py [hunk 3]
```diff
@@ -564,7 +627,29 @@ def model_validator(
...
+        """
+        Decorator that applies the `@classmethod` decorator to a function and returns a
+        `PydanticDescriptorProxy` object.
+
+        Args:
+            f (Any): The function to be decorated.
+
+        Returns:
+            _decorators.PydanticDescriptorProxy[Any]: A PydanticDescriptorProxy object.
+
```
Comment (Kludex) [inline pydantic/decorators.py:651]:
```suggestion
```

**Review by Kludex — APPROVED** [review]

File: pydantic/decorators.py (hunk 1 above)
Comment (Kludex) [reply pydantic/decorators.py:172]: Actually, the description is already on the upper function. I think it's fine.

Comment (tpdorsey) [reply pydantic/decorators.py]: Good callout, and a situation where the coverage tools are in opposition to linting and doc generation tools. I removed the unnecessary docstrings on the `dec` functions.

**Pull request merged**